tests/cloudpickle_test.py

```diff
@@ -1983,6 +1983,134 @@ class A:
         """.format(protocol=self.protocol)
         assert_run_python_script(code)
 
+    def test_dynamic_func_determinist(self):
+        # Check that the pickle serialization for a dynamic func is the same
+        # in two processes.
+
+        def get_dynamic_func_pickle():
+            def test_method(arg_1, arg_2):
+                pass
+            return cloudpickle.dumps(test_method)
+
+        with subprocess_worker(protocol=self.protocol) as w:
+
+            A_dump = w.run(get_dynamic_func_pickle)
+            check_determinist_pickle(A_dump, get_dynamic_func_pickle())
+
+    def test_dynamic_class_determinist(self):
+        # Check that the pickle serialization for a dynamic class is the same
+        # in two processes.
+        pytest.xfail("This test fails due to different tracker_id.")
+
+        def get_dynamic_class_pickle():
+            class A:
+                """Class with potential string interning issues."""
+
+                arg_1 = "class_value"
+
+                def join(self):
+                    pass
+
+                def test_method(self, arg_1, join):
+                    pass
+            return cloudpickle.dumps(A)
+
+        with subprocess_worker(protocol=self.protocol) as w:
+
+            A_dump = w.run(get_dynamic_class_pickle)
+            check_determinist_pickle(A_dump, get_dynamic_class_pickle())
+
+    def test_dynamic_class_determinist_subworker_order(self):
+        # Check that the pickle produced by the unpickled instance is the same.
+        # This checks that the order of the class attributes is deterministic.
+
+        with subprocess_worker(protocol=self.protocol) as w:
+
+            class A:
+                """Simple class definition"""
+
+                pass
+
+            A_dump = w.run(cloudpickle.dumps, A)
+            check_determinist_pickle(A_dump, cloudpickle.dumps(A))
+
+            # If the doc is defined after some class variables, this can cause ordering changes
+            # due to the way we reconstruct the class with _make_class_skeleton, which create
+            # the class and thus `__doc__` before populating it.
```

Comment:
```suggestion
            # If the `__doc__` attribute is defined after some other class attribute,
            # this can cause class attribute ordering changes due to the way we reconstruct
            # the class definition in `_make_class_skeleton`, which creates the class
            # and thus its `__doc__` attribute before populating the class attributes.
```